### PR TITLE
Internal: Editor V2 Preview Button [ED-9725]

### PR DESCRIPTION
**Pull request by rotemee**

### CODE REVIEW

File: packages/packages/top-bar/src/components/locations/main-menu-location.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default function MainMenuLocation() {
 	} );
 
 	return (
-		<Stack sx={ { px: 4 } } direction="row" alignItems="center">
+		<Stack sx={ { paddingInlineStart: 4 } } direction="row" alignItems="center">
```
Comment (Nevoss) [inline packages/packages/top-bar/src/components/locations/main-menu-location.tsx:19]: why? there is no need for padding at the end?

File: packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	isSaving: false,
+	isSavingDraft: false,
+	userCan: {
+		publish: true,
+	},
+};
+
+jest.mock( '../use-active-document.ts', () => ( {
+	__esModule: true,
+	default: jest.fn( () => mockDocument ),
```
Comment (Nevoss) [inline packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts:19]: Maybe use the `createMockDocument` util?

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: packages/packages/top-bar/src/components/locations/main-menu-location.tsx (hunk 1 above)
Comment (rotemee) [reply packages/packages/top-bar/src/components/locations/main-menu-location.tsx:19]: it comes from the tools section which has the same element as the utilities section and both of them needs the padding from the sides..

File: packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts (hunk 2 above)
Comment (rotemee) [reply packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts:10]: Done.

**Review by Nevoss — DISMISSED** [review]

File: packages/packages/documents/src/hooks/use-document-preview-props.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { runCommand } from '@elementor/v1-adapters';
+import useActiveDocument from '../hooks/use-active-document';
+
+export default function useDocumentPreviewProps() {
+	const document = useActiveDocument();
+
+	return {
+		icon: EyeIcon,
+		title: __( 'Preview changes', 'elementor' ),
+		onClick: () => runCommand( 'editor/documents/preview', { id: document?.id } ),
```
Comment (StyleShit) [inline packages/packages/documents/src/hooks/use-document-preview-props.ts:12]: maybe don't run it when there is no document?
and add a test for that

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { renderHook } from '@testing-library/react-hooks';
+import useDocumentPreviewProps from '../use-document-preview-props';
+import { runCommand } from '@elementor/v1-adapters';
+import { createMockDocument } from '../../__tests__/test-utils';
+
+const mockDocument = createMockDocument();
+
+jest.mock( '../use-active-document.ts', () => ( {
```
[the remark this answers is not in the document]
Comment (rotemee) [reply packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts:8]: Done.

File: packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	__esModule: true,
+	default: jest.fn( () => mockDocument ),
+} ) );
+
+jest.mock( '@elementor/v1-adapters', () => ( {
+	runCommand: jest.fn(),
+} ) );
+
+describe( '@elementor/documents - useDocumentPreviewProps', () => {
+	it( 'should trigger the preview command', () => {
```
[the remark this answers is not in the document]
Comment (rotemee) [reply packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts:18]: Done.

File: packages/packages/documents/src/hooks/use-document-preview-props.ts (hunk 3 above)
Comment (rotemee) [reply packages/packages/documents/src/hooks/use-document-preview-props.ts:12]: Done.

File: packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { renderHook } from '@testing-library/react-hooks';
+import useDocumentPreviewProps from '../use-document-preview-props';
+import { runCommand } from '@elementor/v1-adapters';
+import { createMockDocument } from '../../__tests__/test-utils';
+
+const mockDocument = createMockDocument();
```
[the remark this answers is not in the document]
Comment (rotemee) [reply packages/packages/documents/src/hooks/__tests__/use-document-preview-props.test.ts:6]: As we've talked about, creating the mock at the test level.

File: packages/packages/documents/src/icons/eye-icon.tsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { SvgIcon, SvgIconProps } from '@elementor/ui';
+
+export function EyeIcon( props: SvgIconProps ) {
```
[the remark this answers is not in the document]
Comment (rotemee) [reply packages/packages/documents/src/icons/eye-icon.tsx:4]: Done.

File: packages/packages/documents/src/init.ts [hunk 8]
```diff
@@ -17,3 +18,8 @@ injectIntoPrimaryAction( {
 	name: 'top-bar-primary-action',
 	filler: PrimaryAction,
 } );
+
+registerAction( 'utilities', {
+	name: 'top-bar-document-preview',
```
[the remark this answers is not in the document]
Comment (rotemee) [reply packages/packages/documents/src/init.ts]: Done.

**Review by StyleShit — APPROVED** [review]

**Pull request merged**